### PR TITLE
chore: Add remote AMP for Grafana

### PR DESCRIPTION
**Pull request by ovaleanu**
### What does this PR do?

- Add as an option remote AMP to Grafana as additional datastore
- Adds IAM policy and for Grafana to use AMP
- Refactor helm values for kube-prometheus-stack

### Motivation

Add as an option remote AMP to Grafana as additional datastore

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: ai-ml/emr-spark-rapids/addons.tf [hunk 1]
```diff
@@ -18,9 +18,9 @@ module "ebs_csi_driver_irsa" {
 #---------------------------------------------------------------
 # EKS Blueprints Addons
 #---------------------------------------------------------------
-module "eks_blueprints_addons" {
```
Comment (vara-bonthu) [inline ai-ml/emr-spark-rapids/addons.tf:21]: Keep this as `eks_blueprints_addons`

File: ai-ml/emr-spark-rapids/addons.tf [hunk 2]
```diff
@@ -18,9 +18,9 @@ module "ebs_csi_driver_irsa" {
...
+module "eks_blueprints_kubernetes_addons" {
   source  = "aws-ia/eks-blueprints-addons/aws"
-  version = "~> 1.2"
```
Comment (vara-bonthu) [inline ai-ml/emr-spark-rapids/addons.tf:23]: Keep the `1.2` version as well

File: ai-ml/emr-spark-rapids/install.sh [hunk 3]
```diff
@@ -9,6 +9,7 @@ targets=(
   "module.vpc_endpoints"
   "module.eks"
   "module.eks_blueprints_addons"
+  "amp_ingest_irsa"
```
Comment (vara-bonthu) [inline ai-ml/emr-spark-rapids/install.sh:12]: remove this if its not needed or add `module. amp_ingest_irsa `

File: ai-ml/emr-spark-rapids/main.tf [hunk 4]
```diff
@@ -31,7 +34,7 @@ module "eks" {
   manage_aws_auth_configmap = true
   aws_auth_roles = [
     {
-      rolearn  = module.eks_blueprints_addons.karpenter.iam_role_arn
```
Comment (vara-bonthu) [inline ai-ml/emr-spark-rapids/main.tf:34]: `eks_blueprints_addons` keep this name for consistency with EKS Blueprints

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the PR @ovaleanu 👍🏼 I left some minor comments.

This is a big PR. Keep an eye on the TF Plan workflow and any issues.

File: ai-ml/emr-spark-rapids/addons.tf (hunk 1 above)
Comment (ovaleanu) [reply ai-ml/emr-spark-rapids/addons.tf]: Checked in all repos and fixed.

File: ai-ml/emr-spark-rapids/cleanup.sh [hunk 5]
```diff
@@ -8,6 +8,7 @@ export AWS_DEFAULT_REGION=$region
 targets=(
   "module.emr_containers"
   "module.kubernetes_data_addons"
+  "amp_ingest_irsa"
```
[the remark this answers is not in the document]
Comment (ovaleanu) [reply ai-ml/emr-spark-rapids/cleanup.sh]: Checked in all repos and fixed.

File: ai-ml/emr-spark-rapids/main.tf (hunk 4 above)
Comment (ovaleanu) [reply ai-ml/emr-spark-rapids/main.tf]: Checked in all repos and fixed.

File: ai-ml/emr-spark-rapids/install.sh (hunk 3 above)
Comment (ovaleanu) [reply ai-ml/emr-spark-rapids/install.sh]: Checked in all repos and fixed.

File: streaming/flink/install.sh [hunk 6]
```diff
@@ -10,6 +10,7 @@ targets=(
   "module.karpenter"
   "module.ebs_csi_driver_irsa"
   "module.eks_blueprints_kubernetes_addons"
```
Comment (vara-bonthu) [inline streaming/flink/install.sh:12]: this line needs to be removed. Look for any other entries in the PR with `eks_blueprints_kubernetes_addons `

**Review by vara-bonthu — COMMENTED** [review]
LGTM! One minor comment

File: streaming/flink/install.sh (hunk 6 above)
Comment (ovaleanu) [reply streaming/flink/install.sh]: Fixed in all repos

**Review by vara-bonthu — APPROVED** [review]
LGTM 🚀  Thanks @ovaleanu

**Pull request merged**